### PR TITLE
[server] Remove obsolete (Theia) User Storage Resource

### PR DESCRIPTION
**Pull request by AlexTugarev**
## Description

<details>
<summary>Summary generated by Copilot</summary>

### <samp>🤖 Generated by Copilot at a67a742</samp>

This pull request removes all the code and dependencies related to the user storage resources feature from the Gitpod server, protocol, and database components. User storage resources were a deprecated feature that allowed users to allocate disk space for their workspaces. This cleanup simplifies the code base and reduces the complexity of the Gitpod services.

</details>

## Related Issue(s)

Fixes #

#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - at-rm-thei055afa09be</li>
	<li><b>🔗 URL</b> - <a href="https://at-rm-thei055afa09be.preview.gitpod-dev.com/workspaces" target="_blank">at-rm-thei055afa09be.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - at-rm-theia-user-storage-resource-gha.14300</li>
</ul>

## Build Options

<details>
<summary>Build</summary>

- [ ] /werft with-werft
      Run the build with werft instead of GHA
- [ ] leeway-no-cache
- [ ] /werft no-test
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

- [ ] /werft publish-to-npm
- [ ] /werft publish-to-jb-marketplace
</details>

<details>
<summary>Installer</summary>

- [ ] analytics=segment
- [ ] with-dedicated-emulation
- [ ] workspace-feature-flags
  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment</summary>

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-gce-vm
      If enabled this will create the environment on GCE infra
- [ ] with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
- [ ] with-monitoring
</details>

/hold

### CODE REVIEW

**Comment by AlexTugarev** [comment]
@akosyakov, does it make sense to remove this Theia legacy? 
I've checked with DB and last entry is from 2021-09-05 23:42:43.741844000.

**Comment by geropl** [comment]
Taking a tour now... :eyes:

File: components/server/src/code-sync/code-sync-service.ts [hunk 1]
```diff
@@ -343,17 +269,7 @@ export class CodeSyncService {
...
-        }
-        if (
-            !resourceRev &&
-            !collection &&
-            (resourceKey === SyncResource.Extensions || resourceKey === SyncResource.Settings)
-        ) {
-            resourceRev = fromTheiaRev;
-        }
+        const resourceRev = (await this.db.getResource(req.user.id, resourceKey, ref, collection))?.rev;
         if (!resourceRev) {
```
Comment (akosyakov) [inline components/server/src/code-sync/code-sync-service.ts:273]: `!resourceRev || resourceRev === fromTheiaRev` ? 

some clients can already cache it?

Comment (akosyakov) [reply components/server/src/code-sync/code-sync-service.ts:273]: although it should fail like that anyway after trying to access content storage

Comment (AlexTugarev) [reply components/server/src/code-sync/code-sync-service.ts:273]: Would be awesome, if you'd double-check this area. I don't want to break users.

Comment (akosyakov) [reply components/server/src/code-sync/code-sync-service.ts:273]: @jeanp413 could you double check please, on first glance it looks alright

**Review by svenefftinge — COMMENTED** [review]
/workspace/gitpod/components/gitpod-db/src/typeorm/entity/db-user-storage-resource.ts is no longer needed as well

**Review by svenefftinge — APPROVED** [review]
There are also the following two files that need to be updated:
/workspace/gitpod/components/gitpod-protocol/go/gitpod-service.go
/workspace/gitpod/components/gitpod-protocol/go/mock.go

not sure if they are manually edited or generated) Looks like generated but it never works for me so I would usally edit them manually 🤷

File: components/server/src/code-sync/code-sync-service.ts (hunk 1 above)
Comment (jeanp413) [reply components/server/src/code-sync/code-sync-service.ts:273]: yep looks good

**Review by jeanp413 — APPROVED** [review]

**Pull request merged**